typing/typemod.ml

```diff
@@ -1816,85 +1945,116 @@ let type_package env m p nl =
   (wrap_constraint env true modl mty Tmodtype_implicit, tl')
 
 (* Fill in the forward declarations *)
+
+let type_open ?used_slot ovf env loc me =
+  type_open_ ?used_slot ?toplevel:None ovf env loc me
+
+let gen_mod_ident me =
+  match me.pmod_desc with
+  | Pmod_functor _ | Pmod_extension _ -> assert false
+  | Pmod_ident _ -> None
+  | _ -> Some(gen_mod_ident ())
```

Comment:
It would be better to have here an exhaustive list of supported forms.

Reply:
done